include/ignition/rendering/SegmentationCamera.hh

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright (C) 2021 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+#ifndef IGNITION_RENDERING_SEGMENTATIONCAMERA_HH_
+#define IGNITION_RENDERING_SEGMENTATIONCAMERA_HH_
+
+#include <functional>
+#include <string>
+
+#include <ignition/common/Event.hh>
+#include <ignition/math/Color.hh>
+#include "ignition/rendering/Camera.hh"
+
+
+namespace ignition
+{
+  namespace rendering
+  {
+    /// \brief Segmentation types for Semantic / Panpoptic segmentation
+    enum class SegmentationType {
+
+      /// \brief Pixels of same label from different items
+      /// have the same color & id.
+      Semantic = 0,
+
+      /// \brief Pixels of same label from different items, have different
+      /// color & id. 1 channel for label id & 2 channels for instance id
+      Panoptic = 1
```

Comment:
The variable names in the enum should all be uppercase - here's an example: https://github.com/ignitionrobotics/ign-rendering/blob/7b9ac811c6f6489935681047987291c4cb938b6b/include/ignition/rendering/ShaderType.hh#L34-L53

You'll also need to update any other repositories that use this enum to be uppercase as well :grimacing: